### DISCUSSION TITLE
Form validation: Submit or Blur

### DISCUSSION THREAD

**Discussion by iOvergaard** [umbraco/Umbraco.UI#640] [Ideas]
Based on the issue posted by @bjarnef and the comment originally posted by @iOvergaard in https://github.com/umbraco/Umbraco-CMS/issues/15074#issuecomment-1792057809, we should have a discussion of when to do validation: blur or submit.

From a holistic perspective, it would be best to make it configurable either on the uui-form or uui-input, or even better on both. Many UI component libraries have implemented it this way to allow for more refined validation based on actual use cases. For instance, on the V13 login screen it makes sense to do on submit, since we don't have very refined server validation. But when implementing validation in Bellissima, we'll potentially have field-by-field validation.

**Comment by bjarnef** [comment]
Something between the blur or submit could be only to show field as invalid, when a field has been filled with a valid (or the form in dirty as known from Angular). And where form submission validates all fields. Then it won't show validating errors when just tabbing through the form fields with keyboard, but first when user actually has interacted with the field.

e.g. something like this:
https://shoelace.style/getting-started/form-controls/#styling-invalid-form-controls
